src/common/arweave/arweave.ts

```diff
@@ -97,6 +97,9 @@ export class Arweave {
         if (data instanceof Uint8Array) {
           return data.byteLength;
         }
+        if (data == undefined) {
```

Comment:
Seems weird to handle the case where data would be undefined, especially considering line 93 defines data to be of type `string | Uint8Array` so as far as TS is concerned, this case should never happen:
<img width="300px" src="https://user-images.githubusercontent.com/17215508/52852826-f4a66600-3119-11e9-8a1c-6d45704703c5.png">

Then the error message in the error we throw underneath reinforces my comment above, so it feels like this is either not the place to handle this or we should update the type and the error message.